### ISSUE TITLE
[Q]in yolov5 can't find validation accuracy

### ISSUE THREAD

**Issue by besbesmany** [wandb/wandb#2425]
I run yolov5 , I confused with many charts
I don't know where is the val accuracy chart
also what are the meaing of "cls_loss" , "box_loss", "obj_loss"
also what are "mAp_:0.5" and "mAp_:0.5:0.95"

which two of them are the training/validation accuracy ?
and
which two of them are the training/validation loss?

![1](https://user-images.githubusercontent.com/35992924/126546244-85ce5c55-636b-49f5-9551-aa59fc366fab.png)
![2](https://user-images.githubusercontent.com/35992924/126546260-daa52142-52d4-4d37-9689-4b7e785f9b33.png)

**Comment by ivangrov** [comment]
Hello! Thanks for your question. 

In this case the validation accuracy is mAP, and the training and validation losses are under the "train" and "val" tabs. mAP is only being shown for the validation dataset. 

In object detectors the accuracy is measured as mean average precision (or mAP). mAP takes into account how well the bounding boxes are fitting the labels as well as how accurate the predicted classes for those bounding boxes are. The numbers 0.5 or 0.5:0.95 are hyperparameters for mean average precision, saying things like "if the intersection over union (another thing that's used to compute the accuracy) is lower than 0.5 then we are not counting this as a match at all". A good thing to read about mAP that I liked is this: https://yanfengliux.medium.com/the-confusing-metrics-of-ap-and-map-for-object-detection-3113ba0386ef. 

As of the losses: cls_loss is "class loss", "box _loss" is box loss, and "objectness loss" is object loss. I think it just measures the loss relative to those 3 tasks: correctly classifying a class, fitting a bounding box, and determining if there is any object at all in those coordinates (which is kind of a useful thing for object detectors). 

I have made 3 videos about YOLOv5 and W&B as a part of the YOLOv5 Series, which you may find useful. 

YOLOv5 and W&B integration overview: https://youtu.be/yyecuhBmLxE
Installing YOLOv5 locally and running inference: https://www.youtube.com/watch?v=gDoMYuyY_qw. 
Collecting and labelling an object detection dataset, and using W&B Artifacts to log the dataset and then visualize it with Wandb Tables in the browser: https://youtu.be/a9Bre0YJ8L8
Fourth video about training is coming up too.

**Comment by besbesmany** [comment]
@ivangrov Thanks so much for your videos it is really helpful 
if validation accuracy is mAP0.5 and mAP0.5:0.95 , where is the training accuracy how to draw it

I am still confused 
I think validation accuracy is metrics/mAP0.5 and metrics/mAP0.5:0.95 
what about training accuracy?
and
which two of them are the training/validation loss?
I have 
train/box_loss, train/cls_loss, train/obj_loss
val/box_loss, val/cls_loss, val/obj_loss

**Comment by vanpelt** [comment]
@besbesmany accuracy is not measured for the training data in the default yolov5 instrumentation as this would slow down training substantially.

`train/box_loss, train/cls_loss, train/obj_loss` are the different loss values for training.
`val/box_loss, val/cls_loss, val/obj_loss` are the different loss values for validation.

YoloV5 has multiple loss values as described by @ivangrov

**Comment by besbesmany** [comment]
ok how to measure training accuracy , can I do that like val accuracy

**Comment by ivangrov** [comment]
Hey @besbesmany, YOLOv5 computes validation mAP (accuracy) and the validation losses automatically every epoch (and then uses that to perform early stopping and select the best weights, for example). 

As @vanpelt is describing, it is not necessary to compute training mAP for every epoch as it would slow down training. Also, because the training losses are just as good of an indication that the model is getting better on the training set. So, YOLOv5 **does not** compute mAP on the training dataset during training. 

If you would like to explore how you can evaluate YOLOv5 and compute mAP accuracy on a specific dataset, maybe this YOLOv5 issue would be a good place to start https://github.com/ultralytics/yolov5/issues/2242.

**Comment by moahaimen** [comment]
hi all i am new in object detection and in yolov5 i am training it in colab but when i see tensorboard charts i get confused how can i read the values of these charts please i need to know the value of each drawing thank you
![image](https://user-images.githubusercontent.com/30658799/174279145-d1cba59c-2fad-41f2-a2c8-a01a57f8545a.png)
****
**
![image](https://user-images.githubusercontent.com/30658799/174279199-1bf577ac-0c68-4fd2-b361-3c487431e53e.png)
**

**Comment by moahaimen** [comment]
I am honored with your answers,
thank you very much, I appreciate it
 I am new in this field I am also a Ph.D. student, I just began the
research a month ago I need your help if you can, I have other questions,
as you said in the letter which I will quote" As of the losses: cls_loss is
"class loss", "box _loss" is box loss, and "objectness loss" is object
loss. I think it just measures the loss relative to those 3 tasks:
correctly classifying a class, fitting a bounding box, and determining if
there is an object at all in those coordinates (which is kind of a useful
thing for object detectors)."
thank you for the explanations but  the case of yolov5
I don't understand the numbers here is it good or bad?
[image: image.png]
so I need to understand that to determine where the overfitting is
happening?

and the graph is a tensor board. How can I read it so I come up with a
better understanding?

I wish I am not bothering you and I am very thankful for you
